### DISCUSSION TITLE
Where is the logic how CODEOWNERS works on auto assigning?

### DISCUSSION THREAD

**Discussion by tisonkun** [community/community#23356] [New to GitHub]
<p>I’d like to ask the situation when PR matches multiple distinguished rules. Said that we have a CODEOWNERS files like:</p>
<pre><code class="lang-auto">*.js @foo @bar
docs/ @bar @baz
</code></pre>
<p>If someone make a PR touches both JS files and docs folder, is GitHub choose only <code>@bar</code> as the reviewer, or it must require two different reviewers, or some other behavior?</p>
<p>And further, I don’t see a detailed spec describes all these cases how CODEOWNERS file would work. Please DO NOT send me this links, they roughly say little about details. If I want to use it, I’d like to understand most of its behaviors.</p>
<ul>
<li><a href="https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners" class="inline-onebox" rel="noopener nofollow ugc">About code owners - GitHub Docs</a></li>
<li><a href="https://docs.github.com/en/organizations/organizing-members-into-teams/managing-code-review-assignment-for-your-team#about-code-review-assignments" class="inline-onebox" rel="noopener nofollow ugc">Managing code review assignment for your team - GitHub Docs</a></li>
</ul>

**Comment by Zerotask** [comment]
<blockquote>
<p>Order is important; the last matching pattern takes the most<br>
precedence.</p>
</blockquote>
<p>That’s a bit unclear documented in the comments of the example file:<br>
<a href="https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners#example-of-a-codeowners-file" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-repository-on-github/about-code-owners#example-of-a-codeowners-file</a></p>
<p>Since GitHub Docs are now OS, you can create a suggestion for this docs.</p>

**Comment by tisonkun** [comment]
<p>What if there is nothing about order, I mean, a PR touch</p>
<pre><code class="lang-auto">root.js
docs/mydoc.md
</code></pre>
<p>It is my case. Possibly a PR may touch a wide range of file and I’d like to merge CODEOWNERS as many as possible, so that less people be notified (disturbed).</p>

**Comment by Zerotask** [comment]
<p>I don’t really understand what you mean.<br>
Just read the comments from the mentioned example file.</p>

**Comment by tisonkun** [comment]
<blockquote>
<p>Order is important; the last matching pattern takes the most precedence.</p>
</blockquote>
<p>works for one sort of matches, but what if I have two sort of matches.</p>

**Comment by micimize** [comment, marked as answer]
<p>My understanding is that if multiple owner rules are matched, an owner from each rule is required to review the PR. So if <code>index.js</code> and <code>docs/index.md</code> are both edited, either <code>@bar</code> or both <code>@foo</code> and <code>@baz</code> must review the PR.</p>
<p>Note that each rule is combined with “or,” not “and”</p>

**Comment by tisonkun** [comment]
<p>Yep. That sounds correct.</p>
<p>I’ve dropped CODEOWNERS file and think this is the solution. Close with a like solution.</p>
<p>Thanks for your advice <a class="mention" href="/micimize">@micimize</a> !</p>
